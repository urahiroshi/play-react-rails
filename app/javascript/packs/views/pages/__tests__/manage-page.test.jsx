import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ManagePage from '../manage-page.jsx'

describe('when questions are undefined', () => {
  it('should show empty element', () => {
    const wrapper = shallow(<ManagePage startGetQuestions={() => {}} />)
    expect(wrapper.matchesElement(<div />)).toBe(true)
  })
})

describe('when questions are not undefined', () => {
  const createComponent = () => (
    <ManagePage questions={[]} startGetQuestions={() => {}} />
  )

  it('should show manage page', () => {
    const wrapper = shallow(createComponent())
    expect(wrapper.is('.manage-page')).toBe(true)
  })

  it('should show snapshot', () => {
    const wrapper = shallow(createComponent())
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
