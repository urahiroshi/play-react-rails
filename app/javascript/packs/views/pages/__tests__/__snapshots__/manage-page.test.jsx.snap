// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when questions are not undefined should show snapshot 1`] = `
<div
  className="manage-page"
>
  <Connect(NewQuestion) />
  <Questions
    questions={Array []}
  />
</div>
`;
